pkgs/tools/networking/linkchecker/default.nix

```diff
@@ -1,64 +1,48 @@
-{ stdenv, lib, fetchFromGitHub, fetchpatch, python2, gettext }:
-let
-  # pin requests version until next release.
-  # see: https://github.com/linkcheck/linkchecker/issues/76
-  python2Packages = (python2.override {
-    packageOverrides = self: super: {   
-      requests = super.requests.overridePythonAttrs(oldAttrs: rec {
-        version = "2.14.2";
-        src = oldAttrs.src.override {
-          inherit version;
-          sha256 = "0lyi82a0ijs1m7k9w1mqwbmq1qjsac35fazx7xqyh8ws76xanx52";
-        };
-      });
-    };
-  }).pkgs;
-in
+{ stdenv, lib, fetchFromGitHub, python2Packages, gettext }:
+
 python2Packages.buildPythonApplication rec {
   pname = "LinkChecker";
-  version = "9.3.1";
-
-  nativeBuildInputs = [ gettext ];
-  pythonPath = (with python2Packages; [
-    requests
-  ]) ++ [ gettext ];
-
-  checkInputs = with python2Packages; [ pytest ];
+  version = "9.4.0a";
 
-  # the original repository is abandoned, development is now happening here:
   src = fetchFromGitHub {
-    owner = "linkcheck";
+    owner = "linkchecker";
     repo = "linkchecker";
     rev = "v${version}";
-    sha256 = "080mv4iwvlsfnm7l9basd6i8p4q8990mdhkwick9s6javrbf1r1d";
+    sha256 = "1vbwl2vb8dyzki27z3sl5yf9dhdd2cpkg10vbgaz868dhpqlshgs";
   };
 
-  # 1. upstream refuses to support ignoring robots.txt
-  # 2. fix build: https://github.com/linkcheck/linkchecker/issues/10
-  patches = 
-    let
-      fix-setup-py = fetchpatch {
-        name = "fix-setup-py.patch";
-        url = https://github.com/linkcheck/linkchecker/commit/e62e630.patch;
-        sha256 = "046q1whg715w2yv33xx6rkj7fspvvz60cl978ax92lnf8j101czx";
-      };
-    in [
-      ./add-no-robots-flag.patch
-      fix-setup-py
-    ];
+  nativeBuildInputs = [ gettext ];
+  propagatedBuildInputs = with python2Packages; [
```

Comment:
```suggestion

  propagatedBuildInputs = with python2Packages; [
```

Reply:
I can't discern the difference here. Can you help me out?

Reply:
`nativeBuildInputs` and `progagatedBuildInputs` are separated with a line of whitespace.